packages/core/src/lib/connection_manager.ts

```diff
@@ -102,9 +103,10 @@ export class ConnectionManager {
 
         this.dialAttemptsForPeer.delete(peerId.toString());
         return;
-      } catch (error) {
-        log(`
-          Error dialing peer ${peerId.toString()}`);
+      } catch (error: any) {
+        lastErrorRecorded = error;
+        log(`Error dialing peer ${peerId.toString()} - ${error.errors}`);
```

Comment:
`any` have no guarantees for `.errors` to be in place.
Let's use `.?` here and/or maybe add proper type.

Reply:
`catch` does not need typing. I believe the type is TypeScript in-built type `Error`.

Also, I would go against string manipulating errors, just let `debug` handle it:

```
log(`Error dialing peer ${peerId.toString()}`, error);
```